tests/dragonfly/replication_test.py

```diff
@@ -1836,3 +1836,64 @@ async def test_replicaof_reject_on_load(df_local_factory, df_seeder_factory):
     await c_replica.close()
     master.stop()
     replica.stop()
+
+
+# note: please be careful if you want to change any of the parameters used in this test.
+# changing parameters without extensive testing may easily lead to weak testing case assertion
+# which means eviction may not get triggered.
+@pytest.mark.asyncio
+@pytest.mark.slow
+async def test_eviction_propagation(df_local_factory, df_seeder_factory):
+    master = df_local_factory.create(
+        proactor_threads=1, cache_mode="true", maxmemory="256mb", enable_heartbeat_eviction="false"
+    )
+    replica = df_local_factory.create(proactor_threads=1)
+    df_local_factory.start_all([master, replica])
+
+    c_master = master.client()
+    c_replica = replica.client()
+
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+    await wait_available_async(c_replica)
+
+    print("Filling the master to approach maxmemory...")
+    seeder = df_seeder_factory.create(
+        port=master.port, keys=100000, val_size=1000, stop_on_failure=False
```

Comment:
you are filling 100K keys and this is making the test slower, if we use less keys bigger value size the test will run much faster
As I wrote in other comment when you start dragonfly with one shard you will have 6720 slots, so you can run seeder with keys=7000 and value_size=40000 this will get us to eviction much faster

Reply:
in the current setting, I observed that the first bunch of evictions happen at around 55K keys inserted by the seeder...